### ISSUE TITLE
Error in pdk release publish

### ISSUE THREAD

**Issue by coreone** [puppetlabs/pdk#1210]
**Describe the bug**
When running `pdk release publish`, a Ruby error occurs when trying to call the `URI` method:

```sh
pdk (INFO): Releasing module-name - from version 4.0.0
pdk (INFO): Uploading tarball to puppet forge...
Traceback (most recent call last):
	10: from /usr/local/bin/pdk:23:in `<main>'
	 9: from /usr/local/bin/pdk:23:in `load'
	 8: from /var/lib/gems/2.7.0/gems/pdk-2.6.1/exe/pdk:6:in `<top (required)>'
	 7: from /var/lib/gems/2.7.0/gems/pdk-2.6.1/lib/pdk/cli.rb:71:in `run'
	 6: from /var/lib/gems/2.7.0/gems/cri-2.15.11/lib/cri/command.rb:316:in `run'
	 5: from /var/lib/gems/2.7.0/gems/cri-2.15.11/lib/cri/command.rb:316:in `run'
	 4: from /var/lib/gems/2.7.0/gems/cri-2.15.11/lib/cri/command.rb:298:in `run'
	 3: from /var/lib/gems/2.7.0/gems/cri-2.15.11/lib/cri/command.rb:362:in `run_this'
	 2: from /var/lib/gems/2.7.0/gems/pdk-2.6.1/lib/pdk/cli/release/publish.rb:47:in `block (2 levels) in <module:CLI>'
	 1: from /var/lib/gems/2.7.0/gems/pdk-2.6.1/lib/pdk/module/release.rb:91:in `run'
/var/lib/gems/2.7.0/gems/pdk-2.6.1/lib/pdk/module/release.rb:158:in `run_publish': undefined method `URI' for #<PDK::Module::Release:0x000055c4eac01340> (NoMethodError)
```

**To Reproduce**

* Download version 2.6.1 of `pdk`
* cd to a directory containing a pdk-configured repository
* Run `pdk release publish --skip-changelog --skip-dependency --skip-documentation --force --forge-token='abc123'`

**Expected behavior**
This particular example should return an access denied error due to the use of a fake forge token. Using a real token should publish the module to the forge.

**Additional context**
- Your PDK installation method: Gemfile (so, this is run with `bundle exec`)
  - (If via Rubygems, please include your Ruby version `ruby -v`): `ruby 2.7.4p191 (2021-07-07 revision a21a3b7d23) [x86_64-linux-gnu]`
- Your PDK version (`pdk --version`): `2.6.1`
- Your operating system / platform: Ubuntu

** To fix the error **
It looks like the only thing needed to fix this issue is to include a `require 'uri'` at the top of the `lib/pdk/module/release.rb` file. When I do that locally in a test instance, running the `pdk release publish --skip-changelog --skip-dependency --skip-documentation --force --forge-token='abc123'` command returns as I would expect:

```sh
pdk (INFO): Releasing module-name - from version 4.0.0
pdk (INFO): Uploading tarball to puppet forge...
pdk (ERROR): Error uploading to Puppet Forge: {"message":"401 Unauthorized","errors":["The access token is not valid."],"error":"invalid_token","error_description":"The access token is not valid."}
```

**Comment by chelnak** [comment]
@coreone interesting find. I'm surprised this hasn't popped up before now.

We will investigate and get back to you shortly.

**Comment by coreone** [comment]
I edited the `release.rb` file and added `require 'uri'` to the top and then ran the command again and it deployed correctly:

```sh
# bundle exec pdk release publish --skip-changelog --skip-dependency --skip-documentation
pdk (INFO): Releasing module-name - from version 4.0.0
pdk (INFO): Uploading tarball to puppet forge...
pdk (INFO): Publish to Forge was successful
```

**Comment by chelnak** [comment]
What I find interesting here is that URI should be available by default.

I'm triaging this right now and cannot replicate.

Just to confirm, are you running PDK as a gem via a modules Gemfile?

**Comment by chelnak** [comment]
Interesting thanks for the additional info.

I'm actually working on removing the code causing you issues at the moment so hopefully this will be resolved in the next release.. however I'd like to get to the bottom of what is happening in your scenario and see if we can find a workaround.

**Comment by chelnak** [comment]
Fixed in #1220
